### PR TITLE
Issue #1686 Further clean ups following multi entity changes

### PR DESCRIPTION
**Pull request by ljacomet**
This PR contains some more code clean up. It might be easier to review per commit.

It also contains the rename of `EhcacheEntity` into `ClusterTierManagerEntity` ... except for messages which I decided to leave as `EhcacheEntityMessage` and `EhcacheEntityResponse`.

### CODE REVIEW

File: clustered/server/src/main/java/org/ehcache/clustered/server/ClusterTierManagerActiveEntity.java [hunk 1]
```diff
@@ -79,8 +79,8 @@
   private final AtomicBoolean reconnectComplete = new AtomicBoolean(true);
```
Comment (albinsuresh) [inline clustered/server/src/main/java/org/ehcache/clustered/server/ClusterTierManagerActiveEntity.java]: Adding this comment here as I'm not able to comment on the truncated lines.

`storeClientMap` can be removed and `trackedClients` set is redundant. `clientIdMap` has that information already.

Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/ClusterTierManagerActiveEntity.java]: Fixing that

**Comment by ljacomet** [comment]
For 1, aligning on `ClusterTier`, looking into 2

**Comment by ljacomet** [comment]
For 2, that's done with the `connectedClients` field

**Pull request merged**